src/query/storage/mock/storage.go

```diff
@@ -146,7 +157,13 @@ func (s *mockStorage) Fetch(
 	s.Lock()
 	defer s.Unlock()
 	s.lastFetchOptions = opts
-	return s.fetchResult.result, s.fetchResult.err
+	idx := s.fetchResult.idx
+	if idx >= len(s.fetchResult.results) {
+		idx = 0
+	}
+
+	s.fetchResult.idx = s.fetchResult.idx + 1
```

Comment:
Hm, I don't quite get what's going on with the math on `idx` here?

Reply:
Oh, mostly just being lazy haha; if there's no more results sent through, it just returns the first one we sent in.

TBH should probably get rid of these handrolled mocks and refactor tests still using them to use regular generated mocks (would also mean we don't need to keep updating them)